catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/app/SetVariableTest.java

```diff
@@ -0,0 +1,137 @@
+import org.catrobat.catroid.uiespresso.testsuites.Cat;
+import org.catrobat.catroid.uiespresso.testsuites.Level;
+import org.catrobat.catroid.uiespresso.util.rules.BaseActivityInstrumentationRule;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.Espresso.pressBack;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.action.ViewActions.typeText;
+import static android.support.test.espresso.assertion.ViewAssertions.doesNotExist;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withChild;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+
+import static org.catrobat.catroid.uiespresso.content.brick.utils.BrickDataInteractionWrapper.onBrickAtPosition;
+import static org.catrobat.catroid.uiespresso.content.brick.utils.BrickTestUtils.deleteItemWithTextFromFormulaEditorDataList;
+import static org.hamcrest.Matchers.allOf;
+
+@RunWith(AndroidJUnit4.class)
+public class SetVariableTest {
+
+	@Rule
+	public BaseActivityInstrumentationRule<SpriteActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(SpriteActivity.class, true, false);
+
+	@Before
+	public void setUp() throws Exception {
+		createProject();
+		baseActivityTestRule.launchActivity(null);
+	}
+
+	public void createProject() {
+		SetVariableBrick setVariableTestBrick1 = new SetVariableBrick(10);
+		SetVariableBrick setVariableTestBrick2 = new SetVariableBrick(0);
+		Script script = BrickTestUtils.createProjectAndGetStartScript("setVariableTest");
+		script.addBrick(setVariableTestBrick1);
+		script.addBrick(setVariableTestBrick2);
+	}
+
+	@Category({Cat.AppUi.class, Level.Smoke.class})
+	@Test
+	public void testCreateNewUserVariableAndDeletion() {
+		String userVariableName = "testVariable1";
+		String secondUserVariableName = "testVariable2";
+
+		onBrickAtPosition(0).checkShowsText(R.string.brick_when_started);
+		onBrickAtPosition(1).checkShowsText(R.string.brick_set_variable);
+		onBrickAtPosition(1).onVariableSpinner(R.id.set_variable_spinner)
+				.performNewVariableInitial(userVariableName);
+		onBrickAtPosition(1).onVariableSpinner(R.id.set_variable_spinner)
+				.performNewVariable(secondUserVariableName);
+		onBrickAtPosition(1).onFormulaTextField(R.id.brick_set_variable_edit_text)
+				.perform(click());
+
+		deleteItemWithTextFromFormulaEditorDataList(userVariableName, BrickTestUtils.UserVariableUsage.IN_USE);
```

Comment:
theres now a formulaEditorWrapper that does all this stuff...

Reply:
ah OK, thanks a lot for adding these 😃
Now I can finally get rid of deleteItemWithTextFromFormulaEditorDataList which delayed the pull request already several times because it was put into wrong files or had a bad signature etc... (since October)